src/material/button/button.ts

```diff
@@ -91,21 +110,30 @@ export class MatButton extends _MatButtonMixinBase
 
   constructor(elementRef: ElementRef,
               private _focusMonitor: FocusMonitor,
-              @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode: string) {
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode: string,
+              @Inject(MAT_BUTTON_DEFAULT_OPTIONS) defaults: MatButtonDefaultOptions) {
     super(elementRef);
+    const element: HTMLElement = elementRef.nativeElement;
+
+    // Set the default button type if not already specified.
+    if (defaults.type &&
```

Comment:
It might be easier to do this with an `Input` and another entry in the `host` object.